common/persistence/client/factory.go

```diff
@@ -18,6 +18,8 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
+//go:generate mockgen -package $GOPACKAGE -source $GOFILE -destination factory_mock.go
```

Comment:
unfortunately this doesn't allow choosing the interfaces.
I only need the `DataStoreFactory`.